tests/qml/tst_GleanEvents.qml

```diff
@@ -0,0 +1,144 @@
+            let eventData = awaitOn(Sample.settingsAboutUsOpened.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_SpeedTestCompletedEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEventWithExtraKeys("speedTestCompleted", {"speed": "Fast"});
+            let eventData = awaitOn(Sample.speedTestCompleted.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_BottomNavigationBarClickEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEventWithExtraKeys("bottomNavigationBarClick", {"bar_button": "ScreenMessaging"});
+            let eventData = awaitOn(Sample.bottomNavigationBarClick.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_AuthenticationErrorEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEventWithExtraKeys("authenticationError", {"reason": "InvalidTOTP"});
+            let eventData = awaitOn(Sample.authenticationError.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_UserChangedEndpointGeoEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEventWithExtraKeys("userChangedEndpointGeo", {"server": "multiHopExitServer"});
+            let eventData = awaitOn(Sample.userChangedEndpointGeo.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_HelpMenuHelpCenterOpenedEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEvent("helpMenuHelpCenterOpened");
+            let eventData = awaitOn(Sample.helpMenuHelpCenterOpened.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_HelpContactSupportOpenedEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEvent("helpContactSupportOpened");
+            let eventData = awaitOn(Sample.helpContactSupportOpened.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_SupportCaseSubmittedEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEvent("supportCaseSubmitted");
+            let eventData = awaitOn(Sample.supportCaseSubmitted.testGetValue());
+            compare(eventData.length, 1);
+        }
+
+        function test_HelpMenuViewLogsOpenedEvent() {
+            awaitOn(Glean.testResetGlean("mozillavpn", true));
+            TestHelper.triggerRecordGleanEvent("helpMenuViewLogsOpened");
+            let eventData = awaitOn(Sample.helpMenuViewLogsOpened.testGetValue());
+            compare(eventData.length, 1);
+        }
+    }
+}
```

Comment:
This is just testing Glean, not the VPN. Ideally we would perform whatever action should trigger a metric recording and then check that it resulted in a metric recording. Also this is testing deprecated Glean. See my other comment on how to test the "new" Glean. Note that the testing APIs are also available from JavaScript. When reading the documentation on the Glean book, refer to the "Firefox Desktop" tab for the correct API.

Reply:
That makes sense, thanks. I'll essentially turn these into functional tests, may not finish that until tomorrow.